.github/workflows/pr-validation.yaml

```
@@ -0,0 +1,80 @@
+permissions:
+  contents: read
+
+name: PR Validation
+
+on: pull_request
+
+env:
+  GO_VERSION: 1.19
+
+jobs:
+  docs_only_check:
+    name: Check for docs-only change
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    outputs:
+      docs_only: ${{ steps.docs_only_check.outputs.docs_only }}
+    steps:
+    - name: Check out code
+      uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b #v3.2.0
+      with:
+        fetch-depth: 2
+    - id: files
+      name: Get changed files
+      uses: tj-actions/changed-files@0626c3f94002c0a9d7491dd7fed7055bbdff6f92 #v35.1.0
+      with:
+        files_ignore: '**.md'
+    - id: docs_only_check
+      if: steps.files.outputs.any_changed != 'true'
+      name: Check for docs-only changes
+      run: echo "docs_only=true" >> $GITHUB_OUTPUT
+
+  tests:
+    name: file-checks
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone the code
+        uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b #v3.2.0
+        with:
+          fetch-depth: 0
```

```suggestion
        uses: actions/checkout@3
```

I think there shouldn't be much need in pinning this action by the commit.
Also, there's no need to run it with custom fetch-depth.

---

So the fetch thing was a copy-paste from the current grafana-operator repo where we needed it to be able to get the submodule for jsonet but it shouldn't  be needed in the new version so I remove it.

The reason why I use the commit sha is due to that I have thoughts about making our repo salsa compliant https://slsa.dev/ and I have copied a number of things from chainguard that knows their stuff around this. But in short it's due to tags can be rewritten but commits can't so we can trust a commit be consistent.

But I have removed the fetch-depth.

---

We can merge this one first, and in @HubertStefanski PR we can fix it.